modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/loader/validation/EqualsAndHashCodeModelValidatorTestCase.java

```diff
@@ -0,0 +1,116 @@
+import org.mule.runtime.extension.api.annotation.param.Parameter;
+import org.mule.runtime.extension.api.exception.IllegalModelDefinitionException;
+import org.mule.runtime.module.extension.api.loader.java.type.ExtensionParameter;
+import org.mule.runtime.module.extension.internal.loader.java.property.CompileTimeModelProperty;
+import org.mule.runtime.module.extension.internal.loader.java.type.property.ExtensionParameterDescriptorModelProperty;
+import org.mule.runtime.module.extension.internal.loader.java.type.runtime.TypeWrapper;
+import org.mule.tck.junit4.AbstractMuleTestCase;
+import org.mule.tck.size.SmallTest;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import static java.util.Arrays.asList;
+import static org.mockito.Answers.RETURNS_DEEP_STUBS;
+import static org.mockito.Mockito.when;
+import static org.mule.test.module.extension.internal.util.ExtensionsTestUtils.*;
+
+@SmallTest
+@RunWith(MockitoJUnitRunner.class)
+public class EqualsAndHashCodeModelValidatorTestCase extends AbstractMuleTestCase {
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    @Mock(answer = RETURNS_DEEP_STUBS, lenient = true)
+    private ExtensionModel extensionModel;
+
+    @Mock(lenient = true)
+    private OperationModel operationModel;
+
+    @Mock(lenient = true)
+    private ParameterModel parameter;
+
+    @Mock
+    private ExtensionParameter extensionParameter;
+
+    private EqualsAndHashCodeModelValidator validator = new EqualsAndHashCodeModelValidator();
+
+    @Before
+    public void before() {
+        when(extensionModel.getModelProperty(CompileTimeModelProperty.class)).thenReturn(Optional.of(new CompileTimeModelProperty()));
+        when(extensionModel.getOperationModels()).thenReturn(asList(operationModel));
+
+        when(parameter.getModelProperty(ExtensionParameterDescriptorModelProperty.class))
+                .thenReturn(Optional.of(new ExtensionParameterDescriptorModelProperty(extensionParameter)));
+
+        mockParameters(operationModel, parameter);
+    }
+
+    @Test
+    public void pojoMustOverrideEqualsAndHashCode() {
+        expectedException.expect(IllegalModelDefinitionException.class);
+        expectedException
+                .expectMessage("Type 'InvalidPojoMustOverrideEqualsAndHashCode' must override equals and hashCode");
+        when(parameter.getType()).thenReturn(TYPE_LOADER.load(InvalidPojoMustOverrideEqualsAndHashCode.class));
+        when(extensionParameter.getType()).thenReturn(new TypeWrapper(InvalidPojoMustOverrideEqualsAndHashCode.class, TYPE_LOADER));
+        validate(extensionModel, validator);
+    }
+
+    @Test
+    public void pojoImplementsEqualsAndHashCode() {
+        when(parameter.getType()).thenReturn(TYPE_LOADER.load(PojoImplementsEqualsAndHashCode.class));
+        when(extensionParameter.getType()).thenReturn(new TypeWrapper(PojoImplementsEqualsAndHashCode.class, TYPE_LOADER));
+        validate(extensionModel, validator);
+    }
+
```

Comment:
I would add tests for:
The type is an interface
The equal and hash are not implemented in the actual class, but in a superclass that is not java.lang.Object
It is not compile time and validation must not fail even if the methods are not overriden

Reply:
I edited this message because I dont know if one of the cases was valid